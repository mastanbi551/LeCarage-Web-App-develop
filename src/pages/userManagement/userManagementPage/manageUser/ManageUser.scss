@import "../../../../styles//variables.scss";

.manage-cars-page {
  text-align: left;

  .add-button {
    margin-bottom: 125px;
  }

  &__fetaures-info {
    padding-top: 30px;
  }

  .input-text-field {
    .pen-icon {
      position: absolute;
      right: 10px;
      top: 10px;
      color: $black;
    }

    svg {
      color: $lightBlack;
    }
  }

  input {
    width: 100%;
    padding: 10px;
  }

  .label {
    @include label;
  }

  .feature-heading {
    @include montserrat16;
    letter-spacing: 0.15px;
    color: $primaryBlack;
  }

  .feature-info {
    padding-bottom: 20px;
  }

  .checkbox-btn {
    color: $black;
    display: flex;
    padding-right: 20px;

    .checkboxlist,
    .checkbox-text {
      display: inline-block;
    }

    .checkbox-text {
      @include input-text;
      line-height: 20px;
      color: $primaryBlack;
      letter-spacing: 0.25px;
    }
  }

  .heading {
    @include SubHeading2;
  }

  .add-button {
    margin-bottom: 20px;
  }
}

.dropdown {
  margin-bottom: 1px !important;
  padding: 0px;
}
