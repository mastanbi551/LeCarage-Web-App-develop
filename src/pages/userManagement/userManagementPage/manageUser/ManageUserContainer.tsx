import React, { useState, useEffect } from 'react';
import { useLocation } from 'react-router-dom';
import { useAppDispatch, useAppSelector } from '../../../../hooks/Hooks';
import {
  addNewUserRequest,
  onViewadminUserRequest,
  reset,
  updateExistingUserRequest,
} from '../userManagementState/UserManagementSlice';
import { ManageUser } from './ManageUser';
import { useNavigate } from 'react-router';
import _ from 'lodash';
import {
  getAllUserTypes,
  getStatusDetails,
  onLoadingStop,
} from '../../../commonStates/CommonSlice';
import { NotFound } from '../../../notFound/NotFound';
import { AccessTypes, responseStatus, routeNames } from '../../../../utils/constants/ConstantStrings';
import { uploadImage } from '../../../../utils';
import { Loader } from '../../../../components';
export const ManageUserContainer = () => {
  const navigate = useNavigate();
  const dispatch = useAppDispatch();
  const location = useLocation();

  const { responseData, isLoading, viewUserIsLoading, viewUserResponseData } =
    useAppSelector((state) => state.userManagementReducer);
  const { statusList, userTypesList } = useAppSelector(
    (state) => state.commonReducer
  );
  const [userInfo, setUserInfo] = useState(
    location.state && location.state.userInfo
  );

  const [isCreate, setIsCreate] = useState(
    location.state && location.state.action === 'create' ? true : false
  );

  const [isEditable, setIsEditable] = useState(
    location.state &&
      location.state.userInfo &&
      location.state.action === 'edit'
      ? true
      : false
  );

  const [isDisable, setIsDisable] = useState(
    location.state &&
      location.state.userInfo &&
      location.state.action === 'view'
      ? true
      : false
  );

  const [statusDropdownData, setStatusDropdownData] = useState<
    { label: string; value: string }[]
  >([]);
  const [userTypesDropdownData, setUserTypesDropdownData] = useState<
    { label: string; value: string }[]
  >([]);

  const [userData, setUserData] = useState({
    email: '',
    firstName: '',
    lastName: '',
    phoneNumber: '',
    photoURL: '',
    uId: '',
    userStatusId: '',
    userTypeId: '',
    password: '',
  });
  const [userDetailsCollapse, setUserDetailsCollapse] = useState(true);
  const [passwordType, setPasswordType] = useState('password');

  const { routes } = useAppSelector((state) => state.sidebarReducer);
  const [access, setAccess] = useState('');

  useEffect(() => {
    if (access === '' && routes) {
      const SubRoutes = routes ? _.get(routes, routeNames.userManagement) : 0;
      if (SubRoutes) {
        const getAccess = _.filter(
          SubRoutes,
          (obj: {
            accessId: string;
            featureAccess: string;
            isParent: boolean;
            label: string;
            parentFeature: string;
            value: string;
          }) => obj.label === routeNames.users
        );
        if (getAccess.length > 0) {
          setAccess(getAccess ? getAccess[0].featureAccess : '');
        }
      }
    }
  }, [access, routes]);

  const [photoURL, setPhotoURL] = useState('');

  const onDrop = async (file: File) => {
    const webUrl = await uploadImage(file);
    if (webUrl !== false) {
      setPhotoURL(webUrl[0]);
    }
  };

  const handlePasswordTypeChange = () => {
    if (passwordType == '') setPasswordType('password');
    else setPasswordType('');
  };

  const backButtonHandler = () => {
    history.back();
  };

  const saveUpdateButtonHandler = (values: {
    uId: string;
    firstname: string;
    lastname: string;
    email: string;
    cellPhoneNumber: string;
    password: string;
    userStatusId: string;
    userTypeId: string;
  }) => {
    if (isEditable) {   
      const params = JSON.stringify({
        uId: values.uId,
        email: values.email,
        firstName: values.firstname,
        lastName: values.lastname,
        phoneNumber: values.cellPhoneNumber,
        password: values.password,
        userStatusId: values.userStatusId,
        userTypeId: values.userTypeId,
        photoURL: photoURL,
      });
      const formatJson = params.replace(/'/g, '\'\'');
      dispatch(updateExistingUserRequest(formatJson));
    } else {
      const params = JSON.stringify({
        email: values.email,
        firstName: values.firstname,
        lastName: values.lastname,
        phoneNumber: values.cellPhoneNumber,
        password: values.password,
        userStatusId: values.userStatusId,
        userTypeId: values.userTypeId,
        photoURL: photoURL,
      });
      const formatJson = params.replace(/'/g, '\'\'');
      dispatch(addNewUserRequest(formatJson));
    }
  };

  const handleUserDetailsCollapse = () => {
    setUserDetailsCollapse(!userDetailsCollapse);
  };

  useEffect(() => {
    if (isLoading) {
      if (responseData.status === responseStatus.success) {
        dispatch(reset());
        navigate('/users');
        dispatch(onLoadingStop());
      } else if (responseData.status === responseStatus.fail) {
        dispatch(onLoadingStop());
      }
    }
  }, [isLoading, responseData]);

  useEffect(() => {
    if (statusList.length === 0) {
      dispatch(getStatusDetails());
    } else {
      setStatusDropdownData(statusList);
    }
  }, [statusList]);

  useEffect(() => {
    if (userTypesList.length === 0) {
      dispatch(getAllUserTypes());
    } else {
      setUserTypesDropdownData(userTypesList);
    }
  }, [userTypesList]);

  useEffect(() => {
    if (userData.uId === '' && userInfo !== '' && (isEditable || isDisable)) {
      const params = {
        userType: _.get(userInfo, 'User Type').replace(/'/g, '\'\''),
        uId: _.get(userInfo, 'User Id').replace(/'/g, '\'\''),
      };
      dispatch(onViewadminUserRequest(params));
    }
  }, [userData]);

  useEffect(() => {
    if (viewUserIsLoading) {
      if (viewUserResponseData && viewUserResponseData.status === responseStatus.success) {
        setUserData(viewUserResponseData.details[0]);
        setPhotoURL(viewUserResponseData.details[0].photoURL);
      }
    }
  }, [viewUserIsLoading, viewUserResponseData]);

  const handleReset = () => {
    isEditable
      ? setPhotoURL(viewUserResponseData.details[0].photoURL)
      : setPhotoURL('');
  };

  if (access !== '') {
    if (
      access === AccessTypes.fullAccess ||
      access === AccessTypes.readOnlyAccess
    ) {
      return (
        <ManageUser
          backButtonHandler={backButtonHandler}
          userData={userData}
          statusDropDownData={statusDropdownData}
          userTypesDropDownData={userTypesDropdownData}
          isDisable={isDisable}
          isEditable={isEditable}
          isCreate={isCreate}
          userDetailsCollapse={userDetailsCollapse}
          handleUserDetailsCollapse={handleUserDetailsCollapse}
          passwordType={passwordType}
          handlePasswordTypeChange={handlePasswordTypeChange}
          saveOrUpdateButtonHandler={saveUpdateButtonHandler}
          onDrop={onDrop}
          photoURL={photoURL}
          handleReset={handleReset}
        />
      );
    } else {
      return <NotFound />;
    }
  } else {
    return <Loader />;
  }
};
