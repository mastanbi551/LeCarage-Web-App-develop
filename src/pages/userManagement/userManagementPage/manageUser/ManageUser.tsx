import React, { useCallback, useState } from 'react';
import {
  InputTextField,
  PageContainer,
  DetailsContainer,
  CardContainer,
  Loader,
} from '../../../../components';
import './ManageUserContainer';
import { ConstantStrings, ImagesPath } from '../../../../utils/constants';
import { Formik } from 'formik';
import {
  CreateUsersPageValidation, EditUsersPageValidation,
} from '../../../../utils/ValidationSchemas';
import { Collapse } from 'reactstrap';
import { HiEye } from 'react-icons/hi';
import { AiOutlineEyeInvisible } from 'react-icons/ai';
import './ManageUser.scss';
import _ from 'lodash';
import moment from 'moment';
import { uploadImage } from '../../../../utils';
interface ManageUserInteraface {
  backButtonHandler: () => void;
  userData: {
    email: string;
    firstName: string;
    lastName: string;
    phoneNumber: string;
    photoURL: string;
    uId: string;
    userStatusId: string,
    userTypeId: string,
    password: string;
  };
  statusDropDownData: { label: string; value: string }[];
  userTypesDropDownData: { label: string; value: string }[];
  isDisable: boolean;
  isEditable: boolean;
  userDetailsCollapse: boolean;
  handleUserDetailsCollapse: () => void;
  passwordType: string;
  handlePasswordTypeChange: () => void;
  isCreate: boolean;
  handleReset:()=>void;
  saveOrUpdateButtonHandler: (values: {
    uId: string;
    firstname: string,
    lastname: string,
    email: string,
    cellPhoneNumber: string,
    password: string,
    userStatusId: string,
    userTypeId: string,
  }) => void;
  onDrop:(e: File) => void;
  photoURL: string,
}

//ManageUserTypes page handles create Usertypes functionality and update Usertypes functionality
export const ManageUser: React.FC<ManageUserInteraface> = ({
  backButtonHandler,
  userData,
  statusDropDownData,
  userTypesDropDownData,
  isDisable,
  isEditable,
  userDetailsCollapse,
  handleUserDetailsCollapse,
  passwordType,
  handlePasswordTypeChange,
  saveOrUpdateButtonHandler,
  isCreate,
  onDrop,
  photoURL,
  handleReset

}) => {

  return (
    <>
      {((isEditable || isDisable) && userData.uId !== '') || isCreate ? (
        <PageContainer
          mainHeadingName='User'
          backIcon={true}
          backIconOnClick={backButtonHandler}
        >
          <Formik
            initialValues={{
              uId: _.get(userData, 'uId') ? _.get(userData, 'uId') : '',
              firstname: _.get(userData, 'firstName') ? _.get(userData, 'firstName') : '',
              lastname: _.get(userData, 'lastName') ? _.get(userData, 'lastName') : '',
              email: _.get(userData, 'email') ? _.get(userData, 'email') : '',
              cellPhoneNumber: _.get(userData, 'phoneNumber') ? _.get(userData, 'phoneNumber') : '',
              password: _.get(userData, 'password') ? _.get(userData, 'password') : '',
              userStatusId: _.get(userData, 'userStatusId') ? _.get(userData, 'userStatusId') : '',
              userTypeId: _.get(userData, 'userTypeId') ? _.get(userData, 'userTypeId') : '',
            }}
            validationSchema={isEditable ? EditUsersPageValidation : CreateUsersPageValidation}
            onSubmit={(values) => {
              saveOrUpdateButtonHandler(values);
            }}
            
          >
            {({
              handleChange,
              handleSubmit,
              values,
              errors,
              resetForm,
              isValid,
              touched,
              handleBlur
            }) => (
              <form className='container-fluid' onSubmit={handleSubmit}>
                <div className='row manage-usertypes-page '>
                  <div className='col-md-8'>
                    <div className='col-md-12'>
                      <DetailsContainer
                        heading={ConstantStrings.manageUser.userDetailsHeading}
                        isCollapsible={true}
                        collapseOnClick={handleUserDetailsCollapse}
                      >
                        <Collapse isOpen={userDetailsCollapse}>
                          <div className='manage-usertypes-page__usertype-info'>
                            <div className='row '>
                              <div className='col-md-6 pdtopbtm'>
                                <label className='label'>
                                  {ConstantStrings.manageUser.firstName}
                                </label>
                                <InputTextField
                                  name='user-type-id-inputField'
                                  fieldName='userTypeId'
                                  id=''
                                  placeholder=''
                                  inputType='text'
                                  requiredOrNot={true}
                                  value={values.firstname}
                                  onChangeText={handleChange('firstname')}
                                  error={errors.firstname}
                                  touched={touched.firstname}
                                  isDisable={isDisable}
                                  onBlur={handleBlur('firstname')}
                                />
                              </div>
                              <div className='col-md-6 pdtopbtm'>
                                <label className='label'>
                                  {ConstantStrings.manageUser.lastName}
                                </label>
                                <InputTextField
                                  name='user-type-name-inputField'
                                  fieldName='usertypename'
                                  id=''
                                  placeholder=''
                                  inputType='text'
                                  requiredOrNot={true}
                                  value={values.lastname}
                                  onChangeText={handleChange('lastname')}
                                  onBlur={handleBlur('lastname')}
                                  error={errors.lastname}
                                  touched={touched.lastname}
                                  isDisable={isDisable}
                                />
                              </div>

                              <div className='col-md-6 pdtopbtm'>
                                <label className='label'>
                                  {ConstantStrings.manageUser.emailAddress}
                                </label>
                                <InputTextField
                                  name='user-type-description-inputField'
                                  fieldName='usertypeDescription'
                                  id=''
                                  placeholder=''
                                  inputType='email'
                                  requiredOrNot={true}
                                  value={values.email}
                                  onChangeText={handleChange('email')}
                                  onBlur={handleBlur('email')}
                                  error={errors.email}
                                  touched={touched.email}
                                  isDisable={isDisable}
                                />
                              </div>
                              <div className='col-md-6 pdtopbtm'>
                                <label className='label'>
                                  {ConstantStrings.manageUser.cellPhoneNumber}
                                </label>
                                <InputTextField
                                  name='email_field'
                                  fieldName='usertypeDescription'
                                  id='email_field'
                                  placeholder=''
                                  inputType='text'
                                  requiredOrNot={true}
                                  value={values.cellPhoneNumber}
                                  onChangeText={handleChange('cellPhoneNumber')}
                                  onBlur={handleBlur('cellPhoneNumber')}
                                  error={errors.cellPhoneNumber}
                                  touched={touched.cellPhoneNumber}
                                  isDisable={isDisable}
                                />
                              </div>

                              <div className='col-md-6 pdtopbtm manage-usertypes-page__usertype-info__password-fields'>
                                <label className='label'>
                                  {ConstantStrings.manageUser.password}
                                </label>
                                <InputTextField
                                  name='user-type-description-inputField password-field'
                                  fieldName='usertypeDescription'
                                  id=''
                                  placeholder=''
                                  inputType={passwordType}
                                  requiredOrNot={true}
                                  value={values.password}
                                  onChangeText={handleChange('password')}
                                  onBlur={handleBlur('password')}
                                  error={errors.password}
                                  touched={touched.password}
                                  isDisable={isDisable || isEditable ? true : false}
                                >
                                  {isDisable === false && isEditable === false ?
                                    <span className='multiple-icons'>
                                      <i
                                        aria-disabled={isDisable}
                                        onClick={() => {
                                          handlePasswordTypeChange();
                                        }}
                                      >
                                        {passwordType ? (
                                          <HiEye />
                                        ) : (
                                          <AiOutlineEyeInvisible />
                                        )}
                                      </i>
                                    </span> : <span className='multiple-icons'>

                                    </span>}
                                </InputTextField>
                              </div>
                            </div>
                          </div>
                        </Collapse>
                      </DetailsContainer>
                    </div>
                  </div>
                  <div className='col-md-4'>
                    <div className='col-md-12'>
                      <CardContainer
                        heading='User Type Actions'

                        isUserImageAvailable={true}
                        isFileUpload={true}
                        photoURL={photoURL}

                        statusDropDownList={statusDropDownData}
                        statusDropDownOnChange={handleChange('userStatusId')}
                        statusDefaultValue={values.userStatusId}
                        errorStatus={errors.userStatusId}
                        touchedStatus={touched.userStatusId}
                        onBlurStatus={handleBlur('userStatusId')}
                        showStatusSelectText={true}

                        typeDorpDownList={userTypesDropDownData}
                        typeDropDownOnChange={handleChange('userTypeId')}
                        typeDefaultValue={values.userTypeId}
                        errorUserType={errors.userTypeId}
                        touchedUserType={touched.userTypeId}
                        onBlurUserType={handleBlur('userTypeId')}
                        showTypeSelectText={true}
                        onSaveAndUpdateButtonOnClick={handleSubmit}
                        onDiscardButtonOnClick={() => {resetForm();
                          handleReset();
                        }}
                        isValid={isValid}
                        isDisable={isDisable}

                        onDrop={onDrop}
                        isEditable={isEditable}
                      />
                    </div>
                  </div>
                </div>
              </form>
            )}
          </Formik>
        </PageContainer>
      ) : (
        <Loader />
      )}
    </>
  );
};
