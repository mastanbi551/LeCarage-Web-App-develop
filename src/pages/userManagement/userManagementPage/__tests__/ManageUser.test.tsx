import React from 'react';
import { render } from '@testing-library/react';
import { ManageUser } from '../manageUser/ManageUser';

describe('ManageUser component', () => {
  const defaultProps = {
    backButtonHandler: jest.fn(),
    userData: {
      email: '',
      firstName: '',
      lastName: '',
      phoneNumber: '',
      photoURL: '',
      uId: '',
      userStatusId: '',
      userTypeId: '',
      password: '',
    },
    statusDropDownData: [],
    userTypesDropDownData: [],
    isDisable: false,
    isEditable: false,
    userDetailsCollapse: false,
    handleUserDetailsCollapse: jest.fn(),
    passwordType: 'password',
    handlePasswordTypeChange: jest.fn(),
    isCreate: true,
    saveOrUpdateButtonHandler: jest.fn(),
  };

  it('should match snapshot', () => {
    const { container } = render(<ManageUser {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
