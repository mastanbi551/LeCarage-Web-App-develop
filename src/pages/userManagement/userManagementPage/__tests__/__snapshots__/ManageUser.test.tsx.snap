// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageUser component should match snapshot 1`] = `
<div
  class="page-container-details"
>
  <div
    class="page-container-details__header-content"
  >
    <svg
      class="back-icon"
      data-testid="back-icon"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="19"
        x2="5"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 19 5 12 12 5"
      />
    </svg>
    <h1
      class="page-container-details__header-content__heading"
    >
      User
    </h1>
  </div>
  <form
    class="container-fluid"
  >
    <div
      class="row manage-usertypes-page "
    >
      <div
        class="col-md-8"
      >
        <div
          class="col-md-12"
        >
          <div
            class="details-container"
          >
            <div
              class="details-container__details-container-header"
            >
              <h1
                class="details-container__details-container-header__heading"
              >
                User Details
              </h1>
              <svg
                class="details-container__details-container-header__collapse-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="collapse"
            >
              <div
                class="manage-usertypes-page__usertype-info"
              >
                <div
                  class="row "
                >
                  <div
                    class="col-md-6 pdtopbtm"
                  >
                    <label
                      class="label"
                    >
                      First Name
                    </label>
                    <div
                      class="input-text-field"
                    >
                      <input
                        autocomplete="new-password"
                        autocorrect="off"
                        class="user-type-id-inputField static"
                        id=""
                        name="userTypeId"
                        placeholder=""
                        required=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="col-md-6 pdtopbtm"
                  >
                    <label
                      class="label"
                    >
                      Last Name
                    </label>
                    <div
                      class="input-text-field"
                    >
                      <input
                        autocomplete="new-password"
                        autocorrect="off"
                        class="user-type-name-inputField static"
                        id=""
                        name="usertypename"
                        placeholder=""
                        required=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="col-md-6 pdtopbtm"
                  >
                    <label
                      class="label"
                    >
                      Email Address
                    </label>
                    <div
                      class="input-text-field"
                    >
                      <input
                        autocomplete="new-password"
                        autocorrect="off"
                        class="user-type-description-inputField static"
                        id=""
                        name="usertypeDescription"
                        placeholder=""
                        required=""
                        type="email"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="col-md-6 pdtopbtm"
                  >
                    <label
                      class="label"
                    >
                      CellPhone Number
                    </label>
                    <div
                      class="input-text-field"
                    >
                      <input
                        autocomplete="new-password"
                        autocorrect="off"
                        class="email_field static"
                        id="email_field"
                        name="usertypeDescription"
                        placeholder=""
                        required=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="col-md-6 pdtopbtm manage-usertypes-page__usertype-info__password-fields"
                  >
                    <label
                      class="label"
                    >
                      Password
                    </label>
                    <div
                      class="input-text-field"
                    >
                      <input
                        autocomplete="new-password"
                        autocorrect="off"
                        class="user-type-description-inputField password-field static"
                        id=""
                        name="usertypeDescription"
                        placeholder=""
                        required=""
                        type="password"
                        value=""
                      />
                      <span
                        class="multiple-icons"
                      >
                        <i
                          aria-disabled="false"
                        >
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 20 20"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M10 12a2 2 0 100-4 2 2 0 000 4z"
                            />
                            <path
                              clip-rule="evenodd"
                              d="M.458 10C1.732 5.943 5.522 3 10 3s8.268 2.943 9.542 7c-1.274 4.057-5.064 7-9.542 7S1.732 14.057.458 10zM14 10a4 4 0 11-8 0 4 4 0 018 0z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </i>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-4"
      >
        <div
          class="col-md-12"
        >
          <div
            class="card-container"
          >
            <div
              class="col-md-12"
            >
              <div
                class="card-container__image-wrapper"
              >
                <div
                  class="card-container__image-container"
                >
                  <img
                    class="card-container__image-container__image"
                    style="object-fit: cover;"
                  />
                  <h1
                    class="card-container__image-container__heading"
                  />
                </div>
              </div>
              <div
                class="card-container__file-upload-container"
              >
                <h1
                  class="card-container__heading"
                >
                  Upload Image
                </h1>
                <div
                  class="outer"
                >
                  <div
                    class="imgload"
                  >
                     
                  </div>
                  <div
                    class="DropZone"
                    role="button"
                    tabindex="0"
                  >
                    Click or drop your file here
                  </div>
                </div>
              </div>
               
            </div>
            <h1
              class="card-container__heading"
            >
              User Type Actions
            </h1>
            <div
              class="dropdown input-text-field"
            >
              <label
                class="dropDown-label"
              >
                Set Status
              </label>
              <select
                class="custom-select status-dropdown"
              >
                <option
                  class="custom-option"
                  value=""
                >
                  Select
                </option>
              </select>
            </div>
            <div
              class="dropdown input-text-field"
            >
              <label
                class="dropDown-label"
              >
                User Type
              </label>
              <select
                class="custom-select status-dropdown"
              >
                <option
                  class="custom-option"
                  value=""
                >
                  Select
                </option>
              </select>
            </div>
            <div
              class="line"
            />
            <button
              class="greenButton w-100"
              type="button"
            >
              Save & Update
            </button>
            <button
              class="add-button w-100"
              type="button"
            >
              Discard
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
